### ISSUE TITLE
Should the Data Generator Always be Built?

### ISSUE THREAD

**Issue by DeveloperPaul123** [jhu-cisst/cisst#58]
I've been working on a portfile for a `cisst` package to be included in [`vcpkg`](https://github.com/Microsoft/vcpkg) and while doing so discovered that even if `CISST_BUILD_APPLICATIONS` is turned off, the data generator application is still built. Is this to be expected? 

The building of the `cisstDataGeneretor` project also initially fails when first building the `cisst` solution because it can't find `cisstRevision.h` when the build first starts. I'd suggest added a dependency via CMake between `cisstDataGenerator` and the `cisstRevision` build target so that it always completes first before trying to build `cisstDataGenerator`.

**Comment by pkazanzides** [comment]
Yes, the data generator is needed even if `CISST_BUILD_APPLICATIONS` is turned off because it is used for some of the cisst libraries (i.e., every library that has a `cdg` file).  Looking now, I see `cdg` files in `cisstMultiTask`, `cisstStereoVision`, and `cisstParameterTypes`, so it seems that you could turn off the data generator if none of those libraries is being compiled. But, I am not sure this is worth it, and the list of libraries that depend on `cisstDataGenerator` could change in the future.

I will add a ticket about the failure due to missing `cisstRevision.h`.

**Comment by pkazanzides** [comment]
I decided to leave this ticket, rather than creating a new one for `cisstRevision.h`

**Comment by DeveloperPaul123** [comment]
@pkazanzides 

Ok that's fine, I can just clean up the build directory when building via `vcpkg` since it doesn't allow for executables to be built and left in the build trees of the packages it manages. 

I agree that checking if `cisstMultiTask`, `cisstStereoVision` and `cisstParameterTypes` are being built in order to disable `cisstDataGenerator` is not worth it. 

If the issue with the missing `cisstRevision.h` can be solved that will also solve my problem.

**Comment by DeveloperPaul123** [comment]
@pkazanzides Was the `cisstRevision.h` issue ever addressed?

**Comment by adeguet1** [comment]
Sorry, I just discovered I had some old messages on github.  I added the dependency.   @DeveloperPaul123 assuming you don't already have this patched on your branch, can you test this?

**Comment by DeveloperPaul123** [comment]
Thanks @adeguet1 I'll give this another try.
